### PR TITLE
Added HIDDEN level to ignores

### PR DESCRIPTION
**Pull request by vague666**

### CODE REVIEW

**Comment by vague666** [comment]
The testing I did worked well but I might've missed something

File: src/core/ignore.h [hunk 1]
```diff
@@ -29,6 +29,7 @@ int ignore_check(SERVER_REC *server, const char *nick, const char *host,
 enum {
 	IGNORE_FIND_PATTERN = 0x01, /* Match the pattern */
 	IGNORE_FIND_NOACT   = 0x02, /* Exclude the targets with NOACT level */
+	IGNORE_FIND_HIDDEN  = 0x03, /* Exclude the targets with HIDDEN level */
```
Comment (ailin-nemui) [inline src/core/ignore.h]: this should be 0x04 (bits go like 1 2 4 8 0x10 0x20 0x40...)

File: src/fe-common/core/fe-ignore.c [hunk 2]
```diff
@@ -157,7 +157,8 @@ static void cmd_ignore(const char *data)
 		g_strsplit(chanarg, ",", -1);
 
 	rec = ignore_find_full(servertag, mask, patternarg, channels,
-			  IGNORE_FIND_PATTERN | ((level & MSGLEVEL_NO_ACT) ? IGNORE_FIND_NOACT : 0));
+			  IGNORE_FIND_PATTERN | ((level & MSGLEVEL_NO_ACT) ? IGNORE_FIND_NOACT : 0) |
```
Comment (ailin-nemui) [inline src/fe-common/core/fe-ignore.c]: this is getting too ugly, we should add a temp variable flags= IGNORE_FIND_PATTERN and ...if (level &...) flags |= ...;

**Comment by vague666** [comment]
I would prefer making the #define ignore_match_level into a function too, but it didn't become too ugly when I added hidden to it

**Comment by ailin-nemui** [comment]
I think what would help there is a `#define FLAG_MSGLEVELS ( MSGLEVEL_NO_ACT  | MSGLEVEL_HIDDEN )`

**Comment by ailin-nemui** [comment]
@irssi/developers

**Comment by dequis** [comment]
What does this do?

**Comment by ailin-nemui** [comment]
I think it allows you to /ignore ..... hidden and then it will hide the messages instead of ignoring them completely

**Comment by vague666** [comment]
Indeed

**Pull request merged**